integration-tests/src/test/kotlin/org/stellar/anchor/platform/test/Sep24Tests.kt

```diff
@@ -10,140 +10,139 @@ import org.springframework.web.util.UriComponentsBuilder
-  companion object {
-    private val jwtService =
-      JwtService(SEP10_JWT_SECRET, SEP24_INTERACTIVE_URL_JWT_SECRET, SEP24_MORE_INFO_URL_JWT_SECRET)
-    fun setup() {
-      sep24Client = Sep24Client(toml.getString("TRANSFER_SERVER_SEP0024"), jwt)
-    }
-    fun `test Sep24 info endpoint`() {
-      printRequest("Calling GET /info")
-      val info = sep24Client.getInfo()
-      JSONAssert.assertEquals(expectedSep24Info, gson.toJson(info), LENIENT)
-    }
+class Sep24Tests(val config: TestConfig, val toml: TomlContent, jwt: String) {
+  private val jwtService: JwtService =
+    JwtService(
+      config.env["secret.sep10.jwt_secret"]!!,
+      config.env["secret.sep24.interactive_url.jwt_secret"]!!,
+      config.env["secret.sep24.more_info_url.jwt_secret"]!!
+    )
 
-    fun `test Sep24 withdraw`() {
-      printRequest("POST /transactions/withdraw/interactive")
-      val withdrawRequest = gson.fromJson(withdrawRequest, HashMap::class.java)
-      val response = sep24Client.withdraw(withdrawRequest as HashMap<String, String>)
-      printResponse("POST /transactions/withdraw/interactive response:", response)
-      savedWithdrawTxn = sep24Client.getTransaction(response.id, "USDC")
-      printResponse(savedWithdrawTxn)
-      JSONAssert.assertEquals(expectedSep24WithdrawResponse, json(savedWithdrawTxn), LENIENT)
-      // check the returning Sep24InteractiveUrlJwt
-      val params = UriComponentsBuilder.fromUriString(response.url).build().queryParams
-      val cipher = params["token"]!![0]
-      val jwt = jwtService.decode(cipher, Sep24InteractiveUrlJwt::class.java)
-      val data = jwt.claims["data"] as Map<String, String>
-      assertEquals(response.id, data["transaction_id"])
-    }
+  private val platformApiClient =
+    PlatformApiClient(AuthHelper.forNone(), config.env["platform.server.url"]!!)
+  private val sep24Client = Sep24Client(toml.getString("TRANSFER_SERVER_SEP0024"), jwt)
 
-    fun `test Sep24 deposit`() {
-      printRequest("POST /transactions/withdraw/interactive")
-      val depositRequest = gson.fromJson(depositRequest, HashMap::class.java)
-      val response = sep24Client.deposit(depositRequest as HashMap<String, String>)
-      printResponse("POST /transactions/deposit/interactive response:", response)
-      savedDepositTxn = sep24Client.getTransaction(response.id, "USDC")
-      printResponse(savedDepositTxn)
-      JSONAssert.assertEquals(expectedSep24DepositResponse, json(savedDepositTxn), LENIENT)
-      // check the returning Sep24InteractiveUrlJwt
-      val params = UriComponentsBuilder.fromUriString(response.url).build().queryParams
-      val cipher = params["token"]!![0]
-      val jwt = jwtService.decode(cipher, Sep24InteractiveUrlJwt::class.java)
-      val data = jwt.claims["data"] as Map<String, String>
-      assertEquals(response.id, data["transaction_id"])
-    }
+  private fun `test Sep24 info endpoint`() {
+    printRequest("Calling GET /info")
+    val info = sep24Client.getInfo()
```

Comment:
Are you going to replace sep24Client with WalletSDK later?

Reply:
Yes, `Sep${number}Client` classes will be deprecated.